### PR TITLE
ddl: fix string truncating when `tiflash/sync-status` returns a very long line

### LINKED ISSUE #32356: TiDB does not process `tiflash/sync-status` from TiFlash correctly
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
ref https://github.com/pingcap/tics/issues/4033

This is because `ComputeTiFlashStatus` reads string with a `bufio.NewReader`, which can truncate long string when `ReadLine`.
This bug has no effect on any previous versions.

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #32356 

Problem Summary:

### What is changed and how it works?
1. Change ReadLine to ReadString
2. Remove loop

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: store/helper/helper.go [hunk 1]
```diff
@@ -1139,31 +1140,43 @@ func GetTiFlashTableIDFromEndKey(endKey string) int64 {
...
 // ComputeTiFlashStatus is helper function for CollectTiFlashStatus.
 func ComputeTiFlashStatus(reader *bufio.Reader, regionReplica *map[int64]int) error {
-	ns, _, _ := reader.ReadLine()
-	n, err := strconv.ParseInt(string(ns), 10, 64)
+	b, err := ioutil.ReadAll(reader)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	for i := int64(0); i < n; i++ {
-		rs, _, _ := reader.ReadLine()
```
Comment (xhebox) [inline store/helper/helper.go:1148]: I think the original loop is more efficient. And you could workaround the truncation by using `rs, isPrefix, _ := reader.ReadLine()`.

Moreover, you could use `ReadBytes`, suggested by the official document:

> ReadLine is a low-level line-reading primitive. Most callers should use ReadBytes('\n') or ReadString('\n') instead or use a Scanner. `

Also, assert the return value of `ReadXXX`

Comment (CalvinNeo) [reply store/helper/helper.go:1148]: I double checked the server side, seems they will only return a text of two lines. 
So I think can switch to `ReadString`.

**Review by xhebox — APPROVED** [review]

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**